src/transaction_builder.js

```diff
@@ -226,6 +226,19 @@ export class TransactionBuilder {
   addOperation(operation) {
     this.operations.push(operation);
     return this;
+  }	
+  
+  /**
+   * Adds an operation to the transaction at a specific index.
+   *
+   * @param {xdr.Operation} operation - The xdr operation object to add, use {@link Operation} static methods.
+   * @param {number} index - The index at which to insert the operation.
+   *
+   * @returns {TransactionBuilder} - The TransactionBuilder instance for method chaining.
+   */
+  addOperationAtIndex(operation, index) {
```

Comment:
This is great! Thanks for making it so thorough. I'd like to suggest we name these `addOperationAt` and `insertOperationAt` (no `Index` suffix) to better align with JavaScript's [`Array.at`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at).